NewsFeed/Common/Views/ScrollableSegmentedControl.swift

```diff
@@ -0,0 +1,228 @@
+        setUp()
+        createButtons()
+        drawSeparatorLine()
+    }
+    
+    public func setItems(_ items: [String]) {
+        self.buttonTitles = items
+        createButtons()
+        setNeedsLayout()
+    }
+    
+    // Layout subviews - set frames
+    override func layoutSubviews() {
+        super.layoutSubviews()
+        
+        scrollView.frame = bounds
+        
+        updateUI()
+        highlightSelectedSegment()
+        calculateAndSetScrollOffset()
+    }
+    
+    // Action methods
+    @objc private func buttonTapped(_ sender: UIButton) {
+        let button = buttonsArray.filter{$0.tag == selectedSegmentIndex}.first
+        
+        selectedSegmentIndex = sender.tag
+        
+        if let previousButton = button {
+            if previousButton.tag != sender.tag {
+                sendActions(for: .valueChanged)
+                previousButton.isSelected = false
+                UIButton.animate(withDuration: 0.3, animations: {
+                    previousButton.transform = CGAffineTransform.identity
+                })
+            }
+        }
+        
+        sender.isSelected = true
+    }
+    
+    // Private
+    // Set up
+    private func setUp() {
+        scrollView.showsHorizontalScrollIndicator = false
+        addSubview(scrollView)
+    }
+    
+    // Create buttons
+    private func createButtons() {
+        for button in buttonsArray {
+            button.removeFromSuperview()
+        }
+        buttonsArray.removeAll()
+        
+        for (index, title) in buttonTitles.enumerated() {
+            let button = UIButton(type: .custom)
+        
+            button.setTitleColor(UIColor.lightGray, for: .normal)
+            button.setTitleColor(UIColor.black, for: .selected)
+            button.addTarget(self, action: #selector(buttonTapped), for: .touchUpInside)
+            button.setTitle(title, for: .normal)
+            
+            button.tag = index
+            
+            buttonsArray.append(button)
+            
+            scrollView.addSubview(button)
+        }
+    }
+    
+    private func updateUI() {
+        var buttonX: CGFloat = buttonMargin
+        var contentSizeWidth: CGFloat = 0
+        
+        // Set button frame
+        for button in buttonsArray {
+            button.transform = CGAffineTransform.identity
+            button.titleLabel?.font = font
+            
+            let buttonSize = calculateButtonSize(title: (button.titleLabel?.text)!, fontAttributes: [NSAttributedStringKey.font: font!])
+            
+            button.frame = CGRect(x: buttonX,
+                                  y: 0,
+                                  width: buttonSize.width,
+                                  height: scrollView.frame.size.height)
+            
+            buttonX += button.frame.size.width + buttonSpacing
+            
+            contentSizeWidth += button.frame.size.width
+        }
+        
+        scrollView.contentSize.width = buttonX
```

Comment:
Looks like you started tracking `contentSizeWidth` which is a good idea, but left it unused (and with unusable/incorrect result value). There is no need for two temporary variables (`buttonX` _and_ `contentSizeWidth`) - one would be enough.

On that note, there is a small mistake with the new approach. If you look carefully, when your loop is completed, `buttonX` will have `buttonSpacing` as the last added value - and you want it to be `buttonMargin` instead. That's why I said `buttonX`'s value was very close, it was not sarcasm :)

However, there are (at least) 2 solutions I can think of, to have the `buttonMargin` as the last added value, which would make `buttonX` correct.